src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -231,6 +232,11 @@ private CompletableFuture<Result<RequestAndRelatedRecords>> checkPolicy(
       return completedFuture(checkPolicyForTitleLevelHold(records));
     }
 
+    if (DELIVERY == request.getfulfillmentPreference()) {
```

Comment:
We still need to fail if request type is not allowed by policy, so you are exiting too early. I think you need to perform this check in `org.folio.circulation.domain.RequestServiceUtility#refuseWhenRequestCannotBeFulfilled`. Maybe making sure that fulfillment preference is `Hold Shelf` before this check:
```
if (!requestPolicy.allowsServicePoint(requestType, request.getPickupServicePointId()))
```